### PR TITLE
Add query to check if tiller containers are accessible. Closes #430

### LINKED ISSUE #430: Add Tiller Deployment (Helm v2) is accessible from within the cluster query for Kubernetes
### Platform
Kubernetes

### Description
Check if any Tiller Deployment container allows access from within the cluster.

### PR DESCRIPTION
**Pull request by joaomartinscx**
Check if any Tiller Deployment resource has containers not listening to localhost. Closes #430

### CODE REVIEW

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    object.get(metadata.labels,"app","undefined") == "helm"
+}
+
+checkMetadata(metadata) {
+    contains(object.get(metadata.labels,"name","undefined"),"tiller")
+}
+##############################################################
+isAccessible(document) = result {
+    container := document.spec.containers[j]
+    result:= evalContainer(container,j,"")
```
Comment (joelsou5a) [inline assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:43]: result:= evalContainer(container,j,"", metadata)

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+checkMetadata(metadata) {
+    object.get(metadata.labels,"app","undefined") == "helm"
+}
+
+checkMetadata(metadata) {
+    contains(object.get(metadata.labels,"name","undefined"),"tiller")
+}
+##############################################################
+isAccessible(document) = result {
+    container := document.spec.containers[j]
```
Comment (joelsou5a) [inline assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:42]: We will add metadata information in order to be able to improve our searchKey

metadata := document.metadata

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    result:= evalContainer(container,j,"")
+}
+
+isAccessibleTemplate(document) = result {
+    container := document.spec.template.spec.containers[j]
+    result:= evalTemplateContainer(container,j)
+}
+
+evalContainer(container,idx,prefix) = result {
```
Comment (joelsou5a) [inline assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:51]: evalContainer(container,idx,prefix, metadata) = result {

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    contains(container.image,"tiller")
+
+    object.get(container,"args","undefined") == "undefined"
+    result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("%sspec.containers",[prefix]),
```
Comment (joelsou5a) [inline assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:57]: "searchKey": 	    sprintf("metadata.name=%s.%sspec.containers",[metadata.name, prefix]),

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                "issueType":		"MissingAttribute",
+                "keyExpectedValue":  sprintf("'%sspec.containers[%d].args' is set", [prefix,idx]),
+                "keyActualValue": 	sprintf("'%sspec.containers[%d].args' is undefined", [prefix,idx])
+              }
+}
+
+evalContainer(container,idx,prefix) = result {
```
Comment (joelsou5a) [inline assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:64]: evalContainer(container,idx,prefix, metadata) = result {

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego (hunk 4 above)
Comment (joelsou5a) [inline assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:57]: "searchKey": 	    sprintf("metadata.name=%s.%sspec.containers",[metadata.name, prefix]),

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                "issueType":		"IncorrectValue",
+                "keyExpectedValue":  sprintf("'%sspec.containers[%d].args' set the container to listen to localhost", [prefix,idx]),
+                "keyActualValue": 	sprintf("'%sspec.containers[%d].args' is not setting the container to lister to localhost", [prefix,idx])
+              }
+}
+
+evalTemplateContainer(container,idx) = result {
+    result:= evalContainer(container,idx,"spec.template.")
+}
+
```
Comment (joelsou5a) [inline assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:81]: maybe remove and use evalContainer and pass the "spec.template" there

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+}
+##############################################################
+isAccessible(document) = result {
+    container := document.spec.containers[j]
+    result:= evalContainer(container,j,"")
+}
+
+isAccessibleTemplate(document) = result {
+    container := document.spec.template.spec.containers[j]
+    result:= evalTemplateContainer(container,j)
```
Comment (joelsou5a) [inline assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:48]: Check comment bellow
result:= evalContainer(container, j, "spec.template", metadata )

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    contains(object.get(metadata.labels,"name","undefined"),"tiller")
+}
+##############################################################
+isAccessible(document) = result {
+    container := document.spec.containers[j]
+    result:= evalContainer(container,j,"")
+}
+
+isAccessibleTemplate(document) = result {
+    container := document.spec.template.spec.containers[j]
```
Comment (joelsou5a) [inline assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:47]: metadata := document.metadata

**Review by joelsou5a — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego (hunk 2 above)
Comment (joaomartinscx) [reply assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:42]: address changes

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego (hunk 8 above)
Comment (joaomartinscx) [reply assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:47]: address changes

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego (hunk 1 above)
Comment (joaomartinscx) [reply assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:43]: address changes

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego (hunk 7 above)
Comment (joaomartinscx) [reply assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:48]: address changes

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego (hunk 3 above)
Comment (joaomartinscx) [reply assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:51]: address changes

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego (hunk 4 above)
Comment (joaomartinscx) [reply assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:57]: "searchKey": sprintf("metadata.name=%s.%sspec.containers",[metadata.name, prefix]),

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego (hunk 5 above)
Comment (joaomartinscx) [reply assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego:64]: address changes

File: assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego (hunk 6 above)
Comment (joaomartinscx) [reply assets/queries/k8s/tiller_deployment_is_accessible_within_cluster/query.rego]: addressed changes

**Review by joelsou5a — APPROVED** [review]

**Pull request merged**